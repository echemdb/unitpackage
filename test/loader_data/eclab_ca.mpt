EC-Lab ASCII FILE
Nb header lines : 62

Chronoamperometry / Chronocoulometry

Run on channel : 1 (SN 0280)
User :
Electrode connection : standard
Ewe ctrl range : min = -2,50 V, max = 2,50 V
Ewe,I filtering : 50 kHz
Safety Limits :
	Do not start on E overload
Channel : Grounded
Acquisition started on : 12/03/2024 11:03:23
Technique started on : 12/03/2024 11:03:23
Saved on :
	File : 240924_FTO-D30,code107_CA.mpr
	Directory : D:\data\date\
	Host : 192.168.11.182
Device : SP-300 (SN 0095)
Address : USB
EC-Lab for windows v11.26 (software)
Internet server v11.25 (firmware)
Command interpretor v11.26 (firmware)
Electrode material :
Initial state :
Electrolyte :
Comments :
Cable : standard
Reference electrode : Ag/AgCl / KCl (3.5M) (0,205 V)
Electrode surface area : 0,500 cm�
Characteristic mass : 0,001 g
Equivalent Weight : 0,000 g/eq.
Density : 0,000 g/cm3
Cycle Definition : Charge/Discharge alternance
Ei (V)              0,500
vs.                 Ref
ti (h:m:s)          22:00:10,0000
Imax                pass
unit Imax           mA
Imin                pass
unit Imin           mA
dQM                 0,000
unit dQM            mA.h
record              I
dI                  5,000
unit dI             �A
dQ                  0,000
unit dQ             mA.h
dt (s)              0,1000
dta (s)             0,1000
E range min (V)     -2,500
E range max (V)     2,500
I Range             Auto
I Range min         Unset
I Range max         Unset
I Range init        Unset
Bandwidth           8
goto Ns'            0
nc cycles           0

mode	ox/red	error	control changes	Ns changes	counter inc.	Ns	time/s	control/V	Ewe/V	I/mA	dQ/C	(Q-Qo)/C	I Range	Q charge/discharge/mA.h	half cycle	Q discharge/mA.h	Q charge/mA.h	Capacity/mA.h	Efficiency/%	cycle number	P/W
2	1	0	0	0	0	0	0,000000000000000E+000	5,0000000E-001	-3,8460135E-001	2,6524925E-005	0,0000000E+000	0,0000000E+000	45	0,000000000000000E+000	0	0,000000000000000E+000	0,000000000000000E+000	0,000000000000000E+000	0,0000000E+000	0,000000000000000E+000	-1,0201522E-008
2	1	0	1	0	0	0	9,719999754452147E-002	4,9996451E-001	5,0055969E-001	2,4524723E-001	8,4408111E-006	8,4408111E-006	40	2,344669762048094E-006	0	0,000000000000000E+000	2,344669762048094E-006	2,344669762048094E-006	0,0000000E+000	0,000000000000000E+000	1,2276087E-004
2	1	0	1	0	0	0	1,123999971605372E-001	4,9996451E-001	5,0059408E-001	2,4020699E-001	1,2128124E-005	1,2128124E-005	40	3,368923242103645E-006	0	0,000000000000000E+000	3,368923242103645E-006	3,368923242103645E-006	0,0000000E+000	0,000000000000000E+000	1,2024619E-004
2	1	0	1	0	0	0	1,305999967007665E-001	4,9996451E-001	5,0058258E-001	2,3517056E-001	1,6450740E-005	1,6450740E-005	40	4,569650021342871E-006	0	0,000000000000000E+000	4,569650021342871E-006	4,569650021342871E-006	0,0000000E+000	0,000000000000000E+000	1,1772228E-004
2	1	0	1	0	0	0	1,533999961247901E-001	4,9996451E-001	5,0058264E-001	2,3014556E-001	2,1751677E-005	2,1751677E-005	40	6,042132554284763E-006	0	0,000000000000000E+000	6,042132554284763E-006	6,042132554284763E-006	0,0000000E+000	0,000000000000000E+000	1,1520687E-004
2	1	0	1	0	0	0	1,821999953972409E-001	4,9996451E-001	5,0059408E-001	2,2507477E-001	2,8304099E-005	2,8304099E-005	40	7,862249731260818E-006	0	0,000000000000000E+000	7,862249731260818E-006	7,862249731260818E-006	0,0000000E+000	0,000000000000000E+000	1,1267110E-004
2	1	0	1	0	0	0	2,209999944170704E-001	4,9996451E-001	5,0054824E-001	2,2005358E-001	3,6939033E-005	3,6939033E-005	40	1,026084242362736E-005	0	0,000000000000000E+000	1,026084242362736E-005	1,026084242362736E-005	0,0000000E+000	0,000000000000000E+000	1,1014743E-004
2	1	0	1	0	0	0	2,745999930630205E-001	4,9996451E-001	5,0059789E-001	2,1501715E-001	4,8593360E-005	4,8593360E-005	40	1,349815546644903E-005	0	0,000000000000000E+000	1,349815546644903E-005	1,349815546644903E-005	0,0000000E+000	0,000000000000000E+000	1,0763713E-004
2	1	0	1	0	0	0	3,479999912087806E-001	4,9996451E-001	5,0053298E-001	2,1001129E-001	6,4180320E-005	6,4180320E-005	40	1,782786663776885E-005	0	0,000000000000000E+000	1,782786663776885E-005	1,782786663776885E-005	0,0000000E+000	0,000000000000000E+000	1,0511758E-004
2	1	0	1	0	0	0	4,479999886825681E-001	4,9996451E-001	5,0056732E-001	2,0529176E-001	8,4928062E-005	8,4928062E-005	40	2,359112841077149E-005	0	0,000000000000000E+000	2,359112841077149E-005	2,359112841077149E-005	0,0000000E+000	0,000000000000000E+000	1,0276234E-004
2	1	0	1	0	0	0	5,479999861563556E-001	4,9996451E-001	5,0055587E-001	2,0193923E-001	1,0527730E-004	1,0527730E-004	40	2,924369457307168E-005	0	0,000000000000000E+000	2,924369457307168E-005	2,924369457307168E-005	0,0000000E+000	0,000000000000000E+000	1,0108187E-004
2	1	0	1	0	0	0	6,479999836301431E-001	4,9996451E-001	5,0060171E-001	1,9943438E-001	1,2533700E-004	1,2533700E-004	40	3,481583310834443E-005	0	0,000000000000000E+000	3,481583310834443E-005	3,481583310834443E-005	0,0000000E+000	0,000000000000000E+000	9,9837191E-005
2	1	0	1	0	0	0	7,479999811039306E-001	4,9996451E-001	5,0056350E-001	1,9745646E-001	1,4517253E-004	1,4517253E-004	40	4,032570157303578E-005	0	0,000000000000000E+000	4,032570157303578E-005	4,032570157303578E-005	0,0000000E+000	0,000000000000000E+000	9,8839497E-005
2	1	0	1	0	0	0	8,479999785777181E-001	4,9996451E-001	5,0058645E-001	1,9588710E-001	1,6483128E-004	1,6483128E-004	40	4,578646540822875E-005	0	0,000000000000000E+000	4,578646540822875E-005	4,578646540822875E-005	0,0000000E+000	0,000000000000000E+000	9,8058430E-005
2	1	0	1	0	0	0	9,479999760515057E-001	4,9996451E-001	5,0062841E-001	1,9454685E-001	1,8434602E-004	1,8434602E-004	40	5,120722764533841E-005	0	0,000000000000000E+000	5,120722764533841E-005	5,120722764533841E-005	0,0000000E+000	0,000000000000000E+000	9,7395678E-005
2	1	0	1	0	0	0	1,047999973525293E+000	4,9996451E-001	5,0056732E-001	1,9347389E-001	2,0374173E-004	2,0374173E-004	40	5,659492469729028E-005	0	0,000000000000000E+000	5,659492469729028E-005	5,659492469729028E-005	0,0000000E+000	0,000000000000000E+000	9,6846707E-005
2	1	0	1	0	0	0	1,147999970999081E+000	4,9996451E-001	5,0060552E-001	1,9258803E-001	2,2303720E-004	2,2303720E-004	40	6,195477908477187E-005	0	0,000000000000000E+000	6,195477908477187E-005	6,195477908477187E-005	0,0000000E+000	0,000000000000000E+000	9,6410629E-005
2	1	0	1	0	0	0	1,247999968472868E+000	4,9996451E-001	5,0060552E-001	1,9180146E-001	2,4224656E-004	2,4224656E-004	40	6,729071174049750E-005	0	0,000000000000000E+000	6,729071174049750E-005	6,729071174049750E-005	0,0000000E+000	0,000000000000000E+000	9,6016869E-005
2	1	0	1	0	0	0	1,347999965946656E+000	4,9996451E-001	5,0053298E-001	1,9110651E-001	2,6138083E-004	2,6138083E-004	40	7,260578665106247E-005	0	0,000000000000000E+000	7,260578665106247E-005	7,260578665106247E-005	0,0000000E+000	0,000000000000000E+000	9,5655116E-005
2	1	0	1	0	0	0	1,447999963420443E+000	4,9996451E-001	5,0055969E-001	1,9048794E-001	2,8044943E-004	2,8044943E-004	40	7,790261911900921E-005	0	0,000000000000000E+000	7,790261911900921E-005	7,790261911900921E-005	0,0000000E+000	0,000000000000000E+000	9,5350581E-005
2	1	0	1	0	0	0	1,547999960894231E+000	4,9996451E-001	5,0059783E-001	1,8989226E-001	2,9946031E-004	2,9946031E-004	40	8,318342022701270E-005	0	0,000000000000000E+000	8,318342022701270E-005	8,318342022701270E-005	0,0000000E+000	0,000000000000000E+000	9,5059659E-005
2	1	0	1	0	0	0	1,647999958368018E+000	4,9996451E-001	5,0059026E-001	1,8943407E-001	3,1841957E-004	3,1841957E-004	40	8,844987961411889E-005	0	0,000000000000000E+000	8,844987961411889E-005	8,844987961411889E-005	0,0000000E+000	0,000000000000000E+000	9,4828851E-005
2	1	0	1	0	0	0	1,747999955841806E+000	4,9996451E-001	5,0052536E-001	1,8906368E-001	3,3733450E-004	3,3733450E-004	40	9,370402646406244E-005	0	0,000000000000000E+000	9,370402646406244E-005	9,370402646406244E-005	0,0000000E+000	0,000000000000000E+000	9,4631170E-005
2	1	0	1	0	0	0	1,847999953315593E+000	4,9996451E-001	5,0057495E-001	1,8863221E-001	3,5620888E-004	3,5620888E-004	40	9,894691174849868E-005	0	0,000000000000000E+000	9,894691174849868E-005	9,894691174849868E-005	0,0000000E+000	0,000000000000000E+000	9,4424555E-005
2	1	0	1	0	0	0	1,947999950789381E+000	4,9996451E-001	5,0052917E-001	1,8825799E-001	3,7504549E-004	3,7504549E-004	40	1,041793034851758E-004	0	0,000000000000000E+000	1,041793034851758E-004	1,041793034851758E-004	0,0000000E+000	0,000000000000000E+000	9,4228613E-005
2	1	0	1	0	0	0	2,047999948263168E+000	4,9996451E-001	5,0060552E-001	1,8792580E-001	3,9384814E-004	3,9384814E-004	40	1,094022607301465E-004	0	0,000000000000000E+000	1,094022607301465E-004	1,094022607301465E-004	0,0000000E+000	0,000000000000000E+000	9,4076691E-005
2	1	0	1	0	0	0	2,147999945736956E+000	4,9996451E-001	5,0060552E-001	1,8759742E-001	4,1261953E-004	4,1261953E-004	40	1,146165353323643E-004	0	0,000000000000000E+000	1,146165353323643E-004	1,146165353323643E-004	0,0000000E+000	0,000000000000000E+000	9,3912306E-005
2	1	0	1	0	0	0	2,247999943210743E+000	4,9996451E-001	5,0061697E-001	1,8736449E-001	4,3136106E-004	4,3136106E-004	40	1,198225153429020E-004	0	0,000000000000000E+000	1,198225153429020E-004	1,198225153429020E-004	0,0000000E+000	0,000000000000000E+000	9,3797840E-005
2	1	0	1	0	0	0	2,347999940684531E+000	4,9996451E-001	5,0057876E-001	1,8703994E-001	4,5007514E-004	4,5007514E-004	40	1,250208717667394E-004	0	0,000000000000000E+000	1,250208717667394E-004	1,250208717667394E-004	0,0000000E+000	0,000000000000000E+000	9,3628223E-005
2	1	0	1	0	0	0	2,447999938158318E+000	4,9996451E-001	5,0057495E-001	1,8681082E-001	4,6876445E-004	4,6876445E-004	40	1,302123483684328E-004	0	0,000000000000000E+000	1,302123483684328E-004	1,302123483684328E-004	0,0000000E+000	0,000000000000000E+000	9,3512819E-005
2	1	0	1	0	0	0	2,547999935632106E+000	4,9996451E-001	5,0055975E-001	1,8657792E-001	4,8742938E-004	4,8742938E-004	40	1,353970502451476E-004	0	0,000000000000000E+000	1,353970502451476E-004	1,353970502451476E-004	0,0000000E+000	0,000000000000000E+000	9,3393392E-005
2	1	0	1	0	0	0	2,647999933105893E+000	4,9996451E-001	5,0055206E-001	1,8634118E-001	5,0607062E-004	5,0607062E-004	40	1,405751714224203E-004	0	0,000000000000000E+000	1,405751714224203E-004	1,405751714224203E-004	0,0000000E+000	0,000000000000000E+000	9,3273462E-005
2	1	0	1	0	0	0	2,747999930579681E+000	4,9996451E-001	5,0061315E-001	1,8615407E-001	5,2468956E-004	5,2468956E-004	40	1,457471059997463E-004	0	0,000000000000000E+000	1,457471059997463E-004	1,457471059997463E-004	0,0000000E+000	0,000000000000000E+000	9,3191178E-005
2	1	0	1	0	0	0	2,847999928053468E+000	4,9996451E-001	5,0059408E-001	1,8588680E-001	5,4328708E-004	5,4328708E-004	40	1,509130720345096E-004	0	0,000000000000000E+000	1,509130720345096E-004	1,509130720345096E-004	0,0000000E+000	0,000000000000000E+000	9,3053830E-005
2	1	0	1	0	0	0	2,947999925527256E+000	4,9996451E-001	5,0055593E-001	1,8574169E-001	5,6186435E-004	5,6186435E-004	40	1,560734341788209E-004	0	0,000000000000000E+000	1,560734341788209E-004	1,560734341788209E-004	0,0000000E+000	0,000000000000000E+000	9,2974107E-005
2	1	0	1	0	0	0	3,047999923001044E+000	4,9996451E-001	5,0057113E-001	1,8556604E-001	5,8042386E-004	5,8042386E-004	40	1,612288443253686E-004	0	0,000000000000000E+000	1,612288443253686E-004	1,612288443253686E-004	0,0000000E+000	0,000000000000000E+000	9,2889000E-005
2	1	0	1	0	0	0	3,147999920474831E+000	4,9996451E-001	5,0057882E-001	1,8538275E-001	5,9896510E-004	5,9896510E-004	40	1,663791857889543E-004	0	0,000000000000000E+000	1,663791857889543E-004	1,663791857889543E-004	0,0000000E+000	0,000000000000000E+000	9,2798677E-005
2	1	0	1	0	0	0	3,247999917948619E+000	4,9996451E-001	5,0056732E-001	1,8523385E-001	6,1748875E-004	6,1748875E-004	40	1,715246535264370E-004	0	0,000000000000000E+000	1,715246535264370E-004	1,715246535264370E-004	0,0000000E+000	0,000000000000000E+000	9,2722010E-005
