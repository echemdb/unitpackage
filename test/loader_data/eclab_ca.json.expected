{
    "resources": [
        {
            "name": "eclab_ca",
            "type": "table",
            "path": "eclab_ca.csv",
            "scheme": "file",
            "format": "csv",
            "mediatype": "text/csv",
            "encoding": "utf-8",
            "schema": {
                "fields": [
                    {
                        "name": "mode",
                        "type": "integer"
                    },
                    {
                        "name": "ox/red",
                        "type": "integer"
                    },
                    {
                        "name": "error",
                        "type": "integer"
                    },
                    {
                        "name": "control changes",
                        "type": "integer"
                    },
                    {
                        "name": "Ns changes",
                        "type": "integer"
                    },
                    {
                        "name": "counter inc.",
                        "type": "integer"
                    },
                    {
                        "name": "Ns",
                        "type": "integer"
                    },
                    {
                        "name": "time/s",
                        "type": "number"
                    },
                    {
                        "name": "control/V",
                        "type": "number"
                    },
                    {
                        "name": "Ewe/V",
                        "type": "number"
                    },
                    {
                        "name": "I/mA",
                        "type": "number"
                    },
                    {
                        "name": "dQ/C",
                        "type": "number"
                    },
                    {
                        "name": "(Q-Qo)/C",
                        "type": "number"
                    },
                    {
                        "name": "I Range",
                        "type": "integer"
                    },
                    {
                        "name": "Q charge/discharge/mA.h",
                        "type": "number"
                    },
                    {
                        "name": "half cycle",
                        "type": "integer"
                    },
                    {
                        "name": "Q discharge/mA.h",
                        "type": "number"
                    },
                    {
                        "name": "Q charge/mA.h",
                        "type": "number"
                    },
                    {
                        "name": "Capacity/mA.h",
                        "type": "number"
                    },
                    {
                        "name": "Efficiency/%",
                        "type": "number"
                    },
                    {
                        "name": "cycle number",
                        "type": "number"
                    },
                    {
                        "name": "P/W",
                        "type": "number"
                    }
                ]
            },
            "metadata": {
                "echemdb": {
                    "experiment": {
                        "type": "BCV",
                        "comment": "intended approach"
                    },
                    "data description": {
                        "type": "raw",
                        "measurement type": "CA",
                        "scan rate": {
                            "comment": "variable"
                        },
                        "potential scale": {
                            "unit": "V",
                            "reference": "RHE"
                        },
                        "current": {
                            "unit": "mA"
                        }
                    },
                    "electrochemical system": {
                        "instrumentation": [
                            {
                                "type": "potentiostat",
                                "supplier": "Biologic VSP-300"
                            }
                        ],
                        "electrolyte": {
                            "type": "aq",
                            "components": [
                                {
                                    "name": "water",
                                    "type": "solvent",
                                    "source": {
                                        "refinement": "Merk Milli-Q IQ 7000"
                                    },
                                    "purity": {
                                        "grade": "ultrapure water",
                                        "total organic carbon": {
                                            "value": 3,
                                            "unit": "ppb"
                                        },
                                        "total ion conductivity": {
                                            "value": 18,
                                            "unit": "MOhm cm"
                                        }
                                    }
                                },
                                {
                                    "name": "HClO4",
                                    "type": "acid",
                                    "source": {
                                        "supplier": "Sigma Aldrich",
                                        "LOT": null
                                    },
                                    "concentration": {
                                        "value": 0.1,
                                        "unit": "mol / l"
                                    },
                                    "purity": {
                                        "grade": "ACS"
                                    }
                                },
                                {
                                    "name": "N2",
                                    "type": "gas",
                                    "purity": {
                                        "grade": "N5.0"
                                    },
                                    "pressure": {
                                        "unit": 1,
                                        "value": "bar"
                                    },
                                    "proportion": {
                                        "value": 100
                                    }
                                },
                                {
                                    "name": "CO",
                                    "type": "gas",
                                    "purity": {
                                        "value": 99.997,
                                        "unit": "percent"
                                    },
                                    "pressure": {
                                        "value": 1,
                                        "unit": "bar"
                                    },
                                    "proportion": {
                                        "value": 100
                                    }
                                }
                            ],
                            "ph": {
                                "value": 1,
                                "uncertainty": 0.1
                            },
                            "temperature": {
                                "value": 298.15,
                                "unit": "K"
                            }
                        },
                        "electrodes": {
                            "configuration": 3,
                            "reference electrode": {
                                "type": "RHE",
                                "source": {
                                    "supplier": "SI Analytics"
                                }
                            },
                            "counter electrode": {
                                "material": "graphite",
                                "crystallographic orientation": "poly",
                                "shape": "rod"
                            },
                            "working electrode": {
                                "material": "Pt",
                                "crystallographic orientation": "poly",
                                "source": {
                                    "supplier": "MaTecK",
                                    "LOT": 18111601
                                },
                                "shape": {
                                    "type": "beat",
                                    "height": {
                                        "value": "unknown",
                                        "unit": "mm"
                                    },
                                    "diameter": {
                                        "value": "unknown",
                                        "unit": "mm"
                                    }
                                },
                                "geometric electrolyte contact area": {
                                    "value": "unknown",
                                    "unit": "cm-2"
                                },
                                "preparation procedure": "procedure"
                            },
                            "electrochemical cell": {
                                "type": "electrochemical glass cell",
                                "components": [
                                    {
                                        "name": "glass",
                                        "part": "cell"
                                    },
                                    {
                                        "name": "Teflon",
                                        "part": "plug valve"
                                    }
                                ],
                                "cell description": "cell description"
                            }
                        }
                    },
                    "curation": {
                        "process": [
                            {
                                "role": "experimentalist",
                                "name": "Max Doe",
                                "orcid": "https://orcid.org/0000-0002-9686-3948",
                                "date": "2022-04-05"
                            }
                        ]
                    }
                }
            }
        }
    ]
}
